### PR TITLE
working BG identifier integration

### PR DESCRIPTION
**Pull request by maxtaco**
- tested by hand

### CODE REVIEW

**Comment by maxtaco** [comment]
@patrickxb? any interest in taking a look?

File: go/service/bg_identifier.go [hunk 1]
```diff
@@ -0,0 +1,126 @@
...
+	uid             keybase1.UID
+	engine          *engine.BackgroundIdentifier
+	stopCh          chan<- struct{}
+	isDead          bool
+	lastFollowerSet uidSet
+}
+
+func RunBackgroundIdentifier(g *libkb.GlobalContext, u keybase1.UID) (*BackgroundIdentifier, error) {
+	ch := make(chan struct{})
+	eng := engine.NewBackgroundIdentifier(g, ch)
```
Comment (patrickxb) [inline go/service/bg_identifier.go]: I thought the ch in the engine was a snoop channel for testing?

Comment (maxtaco) [inline go/service/bg_identifier.go]: one for testing, and one for killing the loop.

File: go/service/bg_identifier.go [hunk 2]
```diff
@@ -0,0 +1,126 @@
...
+	}
+	b.isDead = true
+	b.stopCh <- struct{}{}
+}
+
+func (b *BackgroundIdentifier) Logout() { b.Shutdown() }
+
+func (b *BackgroundIdentifier) Login(u keybase1.UID) (bgi *BackgroundIdentifier, err error) {
+	defer b.G().Trace("BackgroundIdentifier#Login", func() error { return err })()
+	if b.uid.Equal(u) {
```
Comment (patrickxb) [inline go/service/bg_identifier.go]: don't you still need to start it?  keybase login alice, keybase logout, keybase login will get you here and b.isDead will still be true.

File: go/service/bg_identifier.go [hunk 3]
```diff
@@ -0,0 +1,126 @@
...
+		return nil, nil
+	}
+	b.Logout()
+	return RunBackgroundIdentifier(b.G(), u)
+}
+
+func (b *BackgroundIdentifier) HandleUserChanged(uid keybase1.UID) error {
```
Comment (patrickxb) [inline go/service/bg_identifier.go]: b.Lock()?

File: go/service/bg_identifier.go (hunk 2 above)
Comment (maxtaco) [inline go/service/bg_identifier.go]: you'll get a totally new BgIdentifier in that case. They're single use -- once they stop, the are never restarted.

File: go/service/bg_identifier.go (hunk 3 above)
Comment (maxtaco) [inline go/service/bg_identifier.go]: sure i can move the locking around to be a little more conservative.

File: go/service/bg_identifier.go (hunk 2 above)
Comment (maxtaco) [inline go/service/bg_identifier.go]: You're right... Thanks for finding that...

**Pull request merged**